[GStreamer] Trim leading zeroes from track IDs

vivienne-w opened the pull request:
#### 48efea40e9804b2a54bd9ff9f8dee97ec41c6590
<pre>
[GStreamer] Trim leading zeroes from track IDs
<a href="https://bugs.webkit.org/show_bug.cgi?id=268549">https://bugs.webkit.org/show_bug.cgi?id=268549</a>

Reviewed by Xabier Rodriguez-Calvar and Philippe Normand.

Qtdemux inserts leading zeroes when generating stream-ids, this patch
trims them in TrackPrivateBaseGStreamer for a better representation of
their underlying integers.

* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
(WebCore::trimStreamId):
(WebCore::TrackPrivateBaseGStreamer::TrackPrivateBaseGStreamer):
(WebCore::TrackPrivateBaseGStreamer::notifyTrackOfStreamChanged):
(WebCore::TrackPrivateBaseGStreamer::trackIdFromPadStreamStartOrUniqueID):

Canonical link: <a href="https://commits.webkit.org/274469@main">https://commits.webkit.org/274469@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3be515d519ce449c9af4c14e1f0e4ba8a1642956

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39199 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18176 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41552 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41733 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/35101 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41505 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21029 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15507 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32807 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39773 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15299 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33976 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13288 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13260 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34914 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43010 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35594 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35244 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39071 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/14011 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11564 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37304 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15617 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8770 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/15280 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/15103 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eocanha commented on Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
```diff
@@ -290,8 +301,7 @@ void TrackPrivateBaseGStreamer::notifyTrackOfStreamChanged()
     if (!streamId)
         return;
 
-    GST_INFO("Track %d got stream start for stream %s.", m_index, streamId.get());
```
Please, if you have the opportunity before merging the PR, add again this GST_INFO() line to the final patch. I find these log lines useful when having to debug track issues.
In order to do that, since the PR is already approved, you can just include this line in the commit message. No need to ask for an additional review:
```
Reviewed by Xabier Rodriguez-Calvar and Philippe Normand.
```
Thanks!

calvaris replied:
That line should be included automatically by the commit queue.

eocanha replied:
Even after resubmitting a patch with fixes? I thought any r+ granted was reset in that case. That's why I've always filled in the "Reviewed by" line when resubmitting last minute fixes after the patch was approved. Thanks for clarifying!

calvaris replied:
That happened before github, since github it works.

vivienne-w committed:
[GStreamer] Trim leading zeroes from track IDs
https://bugs.webkit.org/show_bug.cgi?id=268549

Reviewed by Xabier Rodriguez-Calvar and Philippe Normand.

Qtdemux inserts leading zeroes when generating stream-ids, this patch
trims them in TrackPrivateBaseGStreamer for a better representation of
their underlying integers.

* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
(WebCore::trimStreamId):
(WebCore::TrackPrivateBaseGStreamer::TrackPrivateBaseGStreamer):
(WebCore::TrackPrivateBaseGStreamer::notifyTrackOfStreamChanged):
(WebCore::TrackPrivateBaseGStreamer::trackIdFromPadStreamStartOrUniqueID):

Canonical link: https://commits.webkit.org/274469@main
